framework/packages/abstract-std/src/account.rs

```diff
@@ -85,7 +85,18 @@ pub mod state {
         Item::new(storage_namespaces::account::CALLING_TO_AS_ADMIN);
     pub const CALLING_TO_AS_ADMIN_WILD_CARD: &str = "calling-to-wild-card";
 
-    // XION temporary state
+    /// XION temporary state. This is used to make sure that the account only has admin rights when authenticated through XION
+    /// If a call originates from the top level owner account address, there are 2 cases within xion:
+    /// - It's a call made from the account directly.
+    ///     In that case, the tx just got through BeforeTx and the admin_auth flag is set during that hook.
+    ///     --> [`AUTH_ADMIN`] should be true
+    /// - It's a call made from the account through another module.
+    ///     In that case, it means that the execute function on the account was executed successfully
+    ///     --> [`AUTH_ADMIN`] flag is not set anymore, because it's removed at the end of each execution
+    ///
+    /// If a module wants to have admin rights, they need to call through the account, but as soon as the account finishes execution, the Auth_Admin flag is set to false, so there is no chance to do admin actions there
+    ///
+    /// This flag can only be set in the beforeTx hook and is always removed in the AfterTx hook
     pub const AUTH_ADMIN: Item<bool> = Item::new(storage_namespaces::account::AUTH_ADMIN);
```

Comment:
Is this storage item only used with the xion feature? If so, should we flag the storage item as well to prevent use elsewhere?

Reply:
We could, but it's in the abstract std package which doesn't have this feature, so that's annoying, should we add a feature on the std package as well as the account to permit that? Seems heavy

Reply:
But it seems appropriate for feature separation

Reply:
Added xion feature